assets/js/src/form_editor/blocks/segment_select/settings.jsx

```diff
@@ -27,12 +27,30 @@ const Preview = ({ segments, updateSegment }) => {
   };
 
   return segments.map((segment) => (
-    <CheckboxControl
-      label={segment.name}
-      checked={!!segment.isChecked}
-      onChange={partial(onCheck, segment.id)}
-      key={segment.id}
-    />
+    <div className="mailpoet-form-segments-settings-list" key={segment.id}>
+      <CheckboxControl
+        label={segment.name}
+        checked={!!segment.isChecked}
+        onChange={partial(onCheck, segment.id)}
+        key={`check-${segment.id}`}
+      />
+      <span
```

Comment:
Maybe we could use [IconButton component](https://developer.wordpress.org/block-editor/components/icon-button/) provided by wordpress/components. Did you try it?